recipes-kernel/linux/linux-boundary_4.14.x.bb

```diff
@@ -0,0 +1,31 @@
+# Adapted from linux-imx.inc, copyright (C) 2013, 2014 O.S. Systems Software LTDA
+# Released under the MIT license (see COPYING.MIT for the terms)
+
+require recipes-kernel/linux/linux-imx.inc
+
+SUMMARY = "Linux kernel for Boundary Devices boards"
+
+SRC_URI = "git://github.com/boundarydevices/linux-imx6.git;branch=${SRCBRANCH} \
+"
+
+LOCALVERSION = "-2.0.0-ga+yocto"
+SRCBRANCH = "boundary-imx_4.14.x_2.0.0_ga"
+SRCREV = "9abee5da4fbc750da202dbba086ff355c0655d54"
+DEPENDS += "lzop-native bc-native"
+COMPATIBLE_MACHINE = "(nitrogen6x|nitrogen6x-lite|nitrogen6sx|nitrogen7|nitrogen8m|nitrogen8mm)"
+
+KERNEL_DEFCONFIG = "boundary_defconfig"
```

Comment:
Just realized that you set KERNEL_DEFCONFIG both here and in the machine conf file.
I'd say to have it here only for now looking like this:
KERNEL_DEFCONFIG ?= "boundary_defconfig"
So that a machine can override it if needed. Then I'd drop KERNEL_DEFCONFIG from nitrogen8m(m) conf files.
The above only is useful if @otavio agrees on using the in-tree defconfig (not sure if there's a rule in Yocto that says we shouldn't).

Reply:
It is not a rule, however, for user friendliness, it is better to have the defconfig on the layer so it is easier to replace it.

Reply:
Then what about checking for ${WORKDIR}/defconfig in do_copy_default_config to allow a user to still have the possibility to have a defconfig in its layer/bbappend.
Would you accept such approach?
Mainly what we are trying to avoid is:
- duplication of maintenance (defconfig both in tree and Yocto layer)
- forgetting to update Yocto defconfig during an update resulting in errors

Reply:
The problem is end user doesn't know what is enabled / disabled nor where it should put it.

That's the reason we've been adding the files to the layer.

Reply:
So does that mean you will not accept the recipe if the defconfig is not there?
We usually recommend our customers to check /proc/config.gz as it provides the full .config of the kernel being run. Otherwise many got confused with defconfig since it doesn't show all the options (dependencies/auto select).

Reply:
I think you guys are the maintainers, so I'd accept it. Even though I deeply disagree with it.

Reply:
Ok thanks! I'll let the maintainer @chrisdimich decide then ;)